lua/ttt2/libraries/eventdata.lua

```diff
@@ -162,3 +162,56 @@ function eventdata.GetPlayerTotalScores()
 
 	return scoreList
 end
+
+---
+-- Lists all events that grant karma to players. Sorts them by steamID64s.
+-- @return table Returns a table of all karma events per player
+-- @realm shared
+function eventdata.GetPlayerKarma()
+	local eventList = events.list
+	local plysKarma = {}
+
+	-- Go table from back to front as only the newest sync is relevant
+	for j = 1, #eventList do
+		local i = #eventList + 1 - j -- reverse table
+		local event = eventList[i]
+
+		if not event:HasKarma() then continue end
+		plysKarma = event:GetKarma()
+
+		break
+	end
```

Comment:
https://www.lua.org/pil/4.3.4.html

```
local count = #eventList

for i = count, count, -1 do
    local event = eventList[i]
-- ...
end
```

is the best solution imo, anyways, there are several better coding patterns than the current one. E.g. localizing `#eventList` that is called EVERY LOOP, or starting with `j = 0` instead of adding a `1` every time into the `i` calculation. It would simplify the code too ^^

Same in the following function.